feat: support HTTP/SaaS function trigger (#372)
asyncapi/glee: docs/functions.md

|---|---|---|
|send|array&lt;[OutboundMessage](#anatomy-of-an-outbound-message)&gt;|A list of outbound messages to send when the processing of the inbound event has finished. All clients subscribed to the given channel/topic will receive the message.
|reply|array&lt;[OutboundMessage](#anatomy-of-an-outbound-message)&gt;|A list of outbound messages to send as a reply when the processing of the inbound event has finished. This is useful when the target of your message is the sender of the inbound event. Note, however, that this only works when you're running Glee as a server. For example, using `reply` when receiving a WebSocket message is fine and the reply will exclusively go to the client that sent the message. However, if you're receiving a message from an MQTT broker, `reply` will work exactly the same way as `send` above, and will send the message to all the clients subscribed to the given channel/topic.

> |invoke|array&lt;[InvokeRequest](#anatomy-of-an-invoke-message)&gt;|A list of requests for reaching an HTTP/HTTPS endpoint. This is useful when you have a software as a service (SaaS) function and want it to be treated as one of Glee's functions.

Reviewer: ```suggestion
|invoke|array&lt;[InvokeRequest](#anatomy-of-an-invoke-message)&gt;|A list of requests for reaching an HTTP/HTTPS endpoint. This is useful when you have a Serverless function and want it to be treated as one of Glee's functions.
```

|channel|string|The channel/topic you want to send the message to. Defaults to `event.channel`, i.e., the same channel as the received event.
|server|string|The server/broker you want to send the message to. Defaults to `event.serverName`, i.e., the same server as the received event.

##### Anatomy of an invoke message

|Attribute|Type|Description|
|---|---|---|
|url|string|The URL of your HTTP/SaaS end-point.
|headers|object&lt;string,string&gt;|The headers/metadata of the message you want to send.
|method|string|The HTTP method that you want your endpoint to be called with.

> |ignoreResponse|boolean|By default, Glee assumes that the response is in JSON format and adheres to the [Functions](#functions) return type structure. If you do not wish for Glee to process the response, you can set the this option to `false`.

Reviewer: ```suggestion
|ignoreResponse|boolean|By default, Glee assumes that the response is in JSON format and adheres to the [Functions](#functions) return type structure. If you do not wish for Glee to process the response, you can set this option to `false`.
```

> > NOTE: Glee uses [got](https://github.com/sindresorhus/got) under the hood. Therefore, you have the ability to personalize your request with any options that [got](https://github.com/sindresorhus/got#documentation) supports.

Reviewer: How? 🤔

Author: clarified in the new commits.

      ...
```

Glee maps the `onHello` operation to the `functions/onHello.js` file.

> ## Using FaaS functions as glee functions

Reviewer: ```suggestion
## Using Serverless functions as Glee functions
```

> Glee can treat your FaaS function as its own by setting `operationId` to your function URL and customizing it with the `x-glee-invoke` extension:

Reviewer: ```suggestion
Glee can treat your Serverless function as its own by setting `operationId` to your function URL and customizing it with the `x-glee-invoke` extension:
```

```yaml
channels:
  hello:
    publish:
      operationId: https://domain.come/path/to/your/function
      x-glee-invoke:
        method: get
        headers:
          Content-Type: application/json

>         ignoreResponse: true # By default, Glee processes the response and raises an error if it does not conform to the format specified by [functions](#functions) return type.

Reviewer: It would be cool to note that this extension is not required. By default, we'll call the URL on `operationId` with the `POST` method, `Content-Type: application/json` header, and the message as the body.

Also, note that we can't send a request body when the method is `GET`. So we have to think what to do in this case. Do we serialize it as query params along with the URL or do we completely avoid sending the message in the body?

Author: Got it. I guess serialising the message makes sense here. but URLs are limited to 2048 characters which I guess can create untraceable errors (working fine for some requests and failing when the length exceeds 2048).
I think we need to respect the REST API rules/conventions here and ignore the message in GET requests.

Reviewer: Yeah, I would do this too. Honestly, I think most people will use POST. If not all.
